### PR TITLE
Use slash / for package symbols

### LINKED ISSUE #1420: Decide on `language` for package symbols
It's annoying that for javacp, package symbols are `JAVA`, but for scalacp/metac package symbols are `SCALA`.

### LINKED ISSUE #1647: Use slash / for package symbols
Currently, SemanticDB package symbols use `.` like for `scala` in `scala.Predef.String#`. This is problematic since it's not possible to distinguish packages from Scala objects like `Predef`. I propose we change the format to be `scala/Predef.String#` instead. The benefit of this strategy is that we can stop persisting package symbols altogether and we can implement lookup for packages with a stateless method like this

```scala
def info(symbol: String): SymbolInformation =
  if (symbol.endsWith("/")) SymbolInformation(symbol, k.PACKAGE)
  else ???
```

We currently have problems like https://github.com/scalameta/scalameta/issues/1420 where package symbols are attributed different languages in metac/scalacp/javacp.

A downside to this change is that it's less natural to write `scala/collection/immutable/List#` than `scala.collection.immutable.List#` and it's breaking with existing code that hardcodes strings. To smoothen the breaking change, it's possible to support `a.b.c.D#` symbols by automatically converting them to `a/b/c/D#` behind the scenes if you know which which symbols are packages.

### LINKED ISSUE #1655: Remove redundant targeted tests
We have accumulated a large number of test cases in `TargetedSuite` that we can remove because

- expect tests already cover those cases in more detailed way
- they're disorganized
- they're annoying to update when expected behavior changes

The targeted tests are still very handy for local development since they provide a tight edit/run/debug cycle (with 2-4 second turnarounds!). We should keep the testing infrastructure for this purpose alone.

### PR DESCRIPTION
**Pull request by olafurpg**
Fixes #1647.
Fixes #1655 (remove redundant unit tests)
Fixes #1420 (package symbol language)

Currently, SemanticDB package symbols use `.` like for `scala` in `scala.Predef.String#`. This is problematic since it's not possible to distinguish packages from Scala objects like `Predef`. This PR follows up on the proposal in #1647 to change the format to be `scala/Predef.String#` instead. The benefit of this strategy is that we no longer need to persist package symbols and we can compute their `SymbolInformation` with a stateless method like this

```scala
def info(symbol: String): SymbolInformation =
  if (symbol.endsWith("/")) SymbolInformation(symbol, k.PACKAGE)
  else ...
```

### CODE REVIEW

File: tests/jvm/src/test/resources/metac-metacp.diff [hunk 1]
```diff
@@ -1078,122 +1054,147 @@ flags.p.package.S#[T]
...
 
 
+======================
+types/Test.C#Either().
+======================
+--- metac
++++ metacp
+       singleType {
+         prefix {
+-        symbol: "scala/util/Either."
```
Comment (olafurpg) [inline tests/jvm/src/test/resources/metac-metacp.diff]: This is the only regression I can see, and the issue is already tracked in https://github.com/scalameta/scalameta/issues/1625#issuecomment-402174890 I will fix that issue later this week.

Comment (olafurpg) [reply tests/jvm/src/test/resources/metac-metacp.diff]: Turns out #1625 was unrelated after all, this regression has been fixed now.

**Comment by olafurpg** [comment]
Seems like my fix for `scala/util.Either.` by removing `hasTermName` introduced a regression

```
scala.meta.internal.scalacp.MissingSymbolException: scala.reflect.macros.blackbox 
```

I will investigate later.

File: semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/SymbolInformationOps.scala [hunk 2]
```diff
@@ -67,7 +67,8 @@ trait SymbolInformationOps { self: Scalacp =>
           }
           val isModuleClass = sym.entry.entryType == 10
           def isScalaObject: Boolean = isModuleClass && !sym.isJavaDefined
-          if (hasTermName || isScalaObject) k.OBJECT
+          if (sym.isPackageAccordingToClasspath) k.PACKAGE
+          else if (hasTermName || isScalaObject) k.OBJECT
```
Comment (olafurpg) [inline semanticdb/metacp/src/main/scala/scala/meta/internal/scalacp/SymbolInformationOps.scala]: You can ignore the commit 0c5c84b03f2a6910376e624b828051afcfebbb72. I got greedy in 1e3a73bf326538ffa53db726b7465f815b097c50 and tried to fix an issue unrelated to this PR.

File: semanticdb/semanticdb3/semanticdb3.md [hunk 3]
```diff
@@ -2377,8 +2377,8 @@ In this section, we describe the Java symbol format.
 
 **Descriptor** is:
   * For `LOCAL`, unsupported.
-  * For `FIELD` or `PACKAGE`, concatenation of its simple name [\[98\]][98]
-    and a dot (`.`).
+  * For `PACKAGE`, concatenation of its encoded name and a forward slash (`/`).
```
Comment (xeno-by) [inline semanticdb/semanticdb3/semanticdb3.md]: It doesn't look like the Java part of the spec defines what's an encoded name.

File: tests/jvm/src/test/resources/metac-metacp.diff [hunk 4]
```diff
@@ -291,13 +291,13 @@ com.javacp.ClassSuffix#Inner#`<init>`().
 -  tag: PUBLIC
 -  symbol: ""
 +  tag: PRIVATE_WITHIN
-+  symbol: "com.javacp."
++  symbol: "com/javacp/"
```
Comment (xeno-by) [inline tests/jvm/src/test/resources/metac-metacp.diff]: Why did this change?

Comment (olafurpg) [reply tests/jvm/src/test/resources/metac-metacp.diff]: The change is expected, it used to be `com.javacp.` and is now `com/javacp/`. Observe that the diff is only one line, the `PRIVATE_WITHIN` diff is from the diff file itself, it's unchanged in this PR.

File: semanticdb/semanticdb3/semanticdb3.md (hunk 3 above)
Comment (olafurpg) [reply semanticdb/semanticdb3/semanticdb3.md]: Done.

File: tests/jvm/src/test/resources/metac-metacp.diff (hunk 4 above)
Comment (xeno-by) [reply tests/jvm/src/test/resources/metac-metacp.diff]: Ah right.

**Pull request merged**